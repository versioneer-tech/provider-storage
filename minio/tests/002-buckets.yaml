---
apiVersion: pkg.internal/v1beta1
kind: Storage
metadata:
  labels:
    crossplane.io/claim-name: alice-002
    crossplane.io/claim-namespace: default
  name: alice-002
  namespace: storage
spec:
  buckets:
  - bucketName: alice-002
    discoverable: false
  - bucketName: alice-shared-002
    discoverable: true
  owner: alice-002
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: policy-owner-alice-002-alice-002, policy-owner-alice-002-alice-shared-002'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-alice-002-alice-002
  generateName: alice-002-
  labels:
    crossplane.io/composite: alice-002
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-002
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Bucket
      metadata:
        annotations:
          storages.pkg.internal/discoverable: "false"
        name: alice-002
      spec:
        providerConfigRef:
          name: provider-minio
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
  readiness:
    policy: SuccessfulCreate
  watch: false
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-alice-002-alice-shared-002
  generateName: alice-002-
  labels:
    crossplane.io/composite: alice-002
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-002
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Bucket
      metadata:
        annotations:
          storages.pkg.internal/discoverable: "true"
        name: alice-shared-002
      spec:
        providerConfigRef:
          name: provider-minio
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
  readiness:
    policy: SuccessfulCreate
  watch: false
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-owner-alice-002-alice-002
  labels:
    crossplane.io/composite: alice-002
  name: policy-owner-alice-002
  namespace: storage
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-002
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Policy
      metadata:
        name: alice-002.owner.alice-002
      spec:
        forProvider:
          rawPolicy: |-
            {
              "Version": "2012-10-17",
              "Statement": [
                {
                  "Effect": "Allow",
                  "Action": [
                    "s3:*"
                  ],
                  "Resource": [
                    "arn:aws:s3:::alice-002",
                    "arn:aws:s3:::alice-002/*"
                  ]
                }
              ]
            }
        providerConfigRef:
          name: provider-minio
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
  readiness:
    policy: SuccessfulCreate
  watch: false
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-owner-alice-002-alice-shared-002
  labels:
    crossplane.io/composite: alice-002
  name: policy-owner-alice-shared-002
  namespace: storage
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-002
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Policy
      metadata:
        name: alice-002.owner.alice-shared-002
      spec:
        forProvider:
          rawPolicy: |-
            {
              "Version": "2012-10-17",
              "Statement": [
                {
                  "Effect": "Allow",
                  "Action": [
                    "s3:*"
                  ],
                  "Resource": [
                    "arn:aws:s3:::alice-shared-002",
                    "arn:aws:s3:::alice-shared-002/*"
                  ]
                }
              ]
            }
        providerConfigRef:
          name: provider-minio
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
  readiness:
    policy: SuccessfulCreate
  watch: false
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-alice-002
  generateName: alice-002-
  labels:
    crossplane.io/composite: alice-002
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-002
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: User
      metadata:
        name: alice-002
      spec:
        forProvider:
          policies:
          - alice-002.owner.alice-002
          - alice-002.owner.alice-shared-002
          userName: alice-002
        providerConfigRef:
          name: provider-minio
        writeConnectionSecretToRef:
          name: alice-002
          namespace: storage
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
  readiness:
    policy: SuccessfulCreate
  watch: false
