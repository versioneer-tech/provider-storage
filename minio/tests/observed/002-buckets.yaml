---
apiVersion: minio.crossplane.io/v1
kind: Bucket
metadata:
  annotations:
    crossplane.io/claim-name: alice-002
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: bucket-alice-002-alice-002
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: minio.crossplane.io/v1
kind: Bucket
metadata:
  annotations:
    crossplane.io/claim-name: alice-002
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: bucket-alice-002-alice-shared-002
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: minio.crossplane.io/v1
kind: Policy
metadata:
  annotations:
    crossplane.io/claim-name: alice-002
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: policy-alice-002.owner.alice-002
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: minio.crossplane.io/v1
kind: Policy
metadata:
  annotations:
    crossplane.io/claim-name: alice-002
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: policy-alice-002.owner.alice-shared-002
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: minio.crossplane.io/v1
kind: User
metadata:
  annotations:
    crossplane.io/claim-name: alice-002
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: user-alice-002
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
