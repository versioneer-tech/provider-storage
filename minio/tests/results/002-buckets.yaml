---
apiVersion: epca.eo/v1beta1
kind: XStorage
metadata:
  name: alice-002
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
---
apiVersion: minio.crossplane.io/v1
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-alice-002-alice-002
    xstorages.epca.eo/discoverable: "false"
  generateName: alice-002-
  labels:
    crossplane.io/composite: alice-002
  ownerReferences:
  - apiVersion: epca.eo/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-002
    uid: ""
spec:
  providerConfigRef:
    name: storage-minio
---
apiVersion: minio.crossplane.io/v1
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-alice-002-alice-shared-002
    xstorages.epca.eo/discoverable: "true"
  generateName: alice-002-
  labels:
    crossplane.io/composite: alice-002
  ownerReferences:
  - apiVersion: epca.eo/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-002
    uid: ""
spec:
  providerConfigRef:
    name: storage-minio
---
apiVersion: minio.crossplane.io/v1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-alice-002.owner.alice-002
  generateName: alice-002-
  labels:
    crossplane.io/composite: alice-002
  ownerReferences:
  - apiVersion: epca.eo/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-002
    uid: ""
spec:
  forProvider:
    rawPolicy: |
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [ "s3:*" ],
            "Resource": [
              "arn:aws:s3:::alice-002",
              "arn:aws:s3:::alice-002/*"
            ]
          }
        ]
      }
  providerConfigRef:
    name: storage-minio
---
apiVersion: minio.crossplane.io/v1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-alice-002.owner.alice-shared-002
  generateName: alice-002-
  labels:
    crossplane.io/composite: alice-002
  ownerReferences:
  - apiVersion: epca.eo/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-002
    uid: ""
spec:
  forProvider:
    rawPolicy: |
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [ "s3:*" ],
            "Resource": [
              "arn:aws:s3:::alice-shared-002",
              "arn:aws:s3:::alice-shared-002/*"
            ]
          }
        ]
      }
  providerConfigRef:
    name: storage-minio
---
apiVersion: minio.crossplane.io/v1
kind: User
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-alice-002
  generateName: alice-002-
  labels:
    crossplane.io/composite: alice-002
  ownerReferences:
  - apiVersion: epca.eo/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-002
    uid: ""
spec:
  forProvider:
    policies:
    - alice-002.owner.alice-002
    - alice-002.owner.alice-shared-002
    userName: alice-002
  providerConfigRef:
    name: storage-minio
  writeConnectionSecretToRef:
    name: alice-002
    namespace: default
