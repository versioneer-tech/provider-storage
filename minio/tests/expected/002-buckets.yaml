---
apiVersion: pkg.internal/v1beta1
kind: Storage
metadata:
  name: s-jeff
spec:
  bucketAccessGrants:
  - bucketName: b-jeff-shared
    grantee: joe
    permission: ReadOnly
  bucketAccessRequests:
  - bucketName: b-joe
    reason: Need access
  buckets:
  - bucketName: b-jeff
  - bucketName: b-jeff-shared
    discoverable: true
  principal: jeff
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: bucket-b-jeff, bucket-b-jeff-shared, observe-jeff-restricted-b-joe,
      and 4 more'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-b-jeff
  labels:
    crossplane.io/composite: s-jeff
  name: bucket-b-jeff
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jeff
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Bucket
      metadata:
        annotations:
          storages.pkg.internal/discoverable: "false"
        name: b-jeff
      spec:
        providerConfigRef:
          name: provider-minio
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-b-jeff-shared
  labels:
    crossplane.io/composite: s-jeff
  name: bucket-b-jeff-shared
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jeff
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Bucket
      metadata:
        annotations:
          storages.pkg.internal/discoverable: "true"
        name: b-jeff-shared
      spec:
        providerConfigRef:
          name: provider-minio
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-jeff-restricted-b-joe
  labels:
    crossplane.io/composite: s-jeff
  name: observe-jeff-restricted-b-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jeff
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Policy
      metadata:
        name: jeff.restricted.b-joe
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-full-b-jeff
  labels:
    crossplane.io/composite: s-jeff
  name: policy-full-b-jeff
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jeff
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Policy
      metadata:
        name: jeff.full.b-jeff
      spec:
        forProvider:
          rawPolicy: |-
            {
              "Version": "2012-10-17",
              "Statement": [
                {
                  "Effect": "Allow",
                  "Action": [
                    "s3:*"
                  ],
                  "Resource": [
                    "arn:aws:s3:::b-jeff",
                    "arn:aws:s3:::b-jeff/*"
                  ]
                }
              ]
            }
        providerConfigRef:
          name: provider-minio
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-full-b-jeff-shared
  labels:
    crossplane.io/composite: s-jeff
  name: policy-full-b-jeff-shared
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jeff
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Policy
      metadata:
        name: jeff.full.b-jeff-shared
      spec:
        forProvider:
          rawPolicy: |-
            {
              "Version": "2012-10-17",
              "Statement": [
                {
                  "Effect": "Allow",
                  "Action": [
                    "s3:*"
                  ],
                  "Resource": [
                    "arn:aws:s3:::b-jeff-shared",
                    "arn:aws:s3:::b-jeff-shared/*"
                  ]
                }
              ]
            }
        providerConfigRef:
          name: provider-minio
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-joe-restricted-b-jeff-shared
  labels:
    crossplane.io/composite: s-jeff
  name: policy-joe-restricted-b-jeff-shared
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jeff
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Policy
      metadata:
        name: joe.restricted.b-jeff-shared
      spec:
        forProvider:
          rawPolicy: |-
            {
              "Version": "2012-10-17",
              "Statement": [
                {
                  "Effect": "Allow",
                  "Action": [
                    "s3:GetBucketLocation",
                    "s3:ListBucket"
                  ],
                  "Resource": [
                    "arn:aws:s3:::b-jeff-shared"
                  ]
                },
                {
                  "Effect": "Allow",
                  "Action": [
                    "s3:GetObject"
                  ],
                  "Resource": [
                    "arn:aws:s3:::b-jeff-shared/*"
                  ]
                }
              ]
            }
        providerConfigRef:
          name: provider-minio
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-jeff
  labels:
    crossplane.io/composite: s-jeff
  name: user-jeff
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jeff
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: User
      metadata:
        name: jeff
      spec:
        forProvider:
          policies:
          - jeff.full.b-jeff
          - jeff.full.b-jeff-shared
          userName: jeff
        providerConfigRef:
          name: provider-minio
        writeConnectionSecretToRef:
          name: jeff
          namespace: default
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
